sky/utils/common_utils.py

```diff
@@ -98,10 +97,12 @@ def _is_valid_user_hash(user_hash: Optional[str]) -> bool:
             return user_hash
 
     hash_str = user_and_hostname_hash()
-    user_hash = hashlib.md5(hash_str.encode()).hexdigest()[:USER_HASH_LENGTH]
-    if not _is_valid_user_hash(user_hash):
+    hex_hash = hashlib.md5(hash_str.encode()).hexdigest()[:USER_HASH_LENGTH]
+    if not _is_valid_user_hash(hex_hash):
         # A fallback in case the hash is invalid.
-        user_hash = uuid.uuid4().hex[:USER_HASH_LENGTH]
+        hex_hash = uuid.uuid4().hex[:USER_HASH_LENGTH]
+    user_hash = base36_encode(hex_hash)
```

Comment:
We should not take the shorter version before the base36 encoding? Otherwise, it will be shorter than `USER_HASH_LENGTH`, right?

Reply:
```python
    hex_hash = hashlib.md5(hash_str.encode()).hexdigest()
    if not _is_valid_user_hash(hex_hash):
        # A fallback in case the hash is invalid.
        hex_hash = uuid.uuid4().hex
    user_hash = base36_encode(hex_hash)[:USER_HASH_LENGTH]
```

Reply:
Actually, since this may inflate the user hash, if a user accidentally delete the user hash file, let's keep it in the original way: 16^8 > 4B space, which is large enough for our users.

Reply:
That's a good point! I reverted in 1bc79d0. The original approach already provides enough hash space, so keeping it makes sense.